packages/typegpu/src/tgsl/wgslGenerator.ts

```diff
@@ -377,7 +377,10 @@ export function generateExpression(
 
       // Either `Struct({ x: 1, y: 2 })`, or `Struct(otherStruct)`.
       // In both cases, we just let the argument resolve everything.
-      return snip(ctx.resolve(arg.value), callee.value);
+      return snip(
+        ctx.resolveValue(arg.value as never, callee.value),
+        callee.value,
+      );
```

Comment:
👉👈
I know it's a little bloated but actually tells us in the future what the value is at a glance
```suggestion
      return snip(
        ctx.resolveValue(
          arg.value as unknown[] | {
            [x: string]: unknown;
          },
          callee.value,
        ),
        callee.value,
      );
```